pallets/providers/src/types.rs

```diff
@@ -0,0 +1,67 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::pallet_prelude::*;
+use frame_support::traits::fungible::Inspect;
+use frame_support::BoundedVec;
+use scale_info::TypeInfo;
+
+use crate::Config;
+
+/// Structure that has the possible value propositions that a main storage provider can offer (and the runtime is aware of)
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, RuntimeDebugNoBound, PartialEq, Eq, Clone)]
+#[scale_info(skip_type_params(T))]
+pub struct ValueProposition<T: Config> {
+    pub data_limit: StorageData<T>,
+    pub protocols: BoundedVec<Protocols<T>, MaxProtocols<T>>,
+    // todo!("add relevant fields here")
+}
+
+/// Structure that represents a main storage provider. It holds the amount of data that the MSP is able to store,
+/// the amount of data that it IS storing, its libp2p multiaddress, and its value proposition.
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, RuntimeDebugNoBound, PartialEq, Eq, Clone)]
+#[scale_info(skip_type_params(T))]
+pub struct MainStorageProvider<T: Config> {
+    pub total_data: StorageData<T>,
+    pub data_used: StorageData<T>,
+    pub multiaddress: MultiAddress<T>,
+    pub value_prop: ValueProposition<T>,
+}
+
+/// Structure that represents a backup storage provider. It holds the amount of data that the BSP is able to store,
+/// the amount of data that it is storing, and its libp2p multiaddress.
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, RuntimeDebugNoBound, PartialEq, Eq, Clone)]
+#[scale_info(skip_type_params(T))]
+pub struct BackupStorageProvider<T: Config> {
+    pub total_data: StorageData<T>,
+    pub data_used: StorageData<T>,
+    pub multiaddress: MultiAddress<T>,
```

Comment:
Should be able to contain multiple multiaddresses. 

Enables BSPs to serve from multiple nodes at once and have the user attempt retry connections until it hits an up and running node, increasing BSP scalability and availability.

Reply:
Working on it!